api/src/main/java/org/openmrs/api/db/hibernate/HibernatePersonDAO.java

```diff
@@ -92,112 +90,67 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 		ListPart<Object[]> names = luceneQuery.listPartProjection(0, maxResults, "person.personId");
 		names.getList().forEach(x -> people.add(getPerson((Integer) x[0])));
 		
-		Set<Person> firstNamesMatch = new HashSet<>();
-		Set<Person> secondNamesMatch = new HashSet<>();
-		Set<Person> secondNamesMatchFirstName = new HashSet<>();
-		Set<Person> firstNamesMatchSecondName = new HashSet<>();
-		
-		
-		// Executed Filter Logic: 
-		// People whose first name matches n1 and second part of the name (givenName, familyName, familyName2) matches at least once n2
-		firstNamesMatch = execSoundexLuceneQueryByNames(true, name1, birthyear, includeVoided, gender);
-		secondNamesMatch = execSoundexLuceneQueryByNames(false, name2, birthyear, includeVoided, gender);
-		
-		firstNamesMatch.retainAll(secondNamesMatch);
-		people.addAll(firstNamesMatch);
-		
-		// Executed Filter Logic: 
-		// People who have second part of the name(givenName, familyName, familyName2) matching n2 and n1 (at least once)
-		secondNamesMatchFirstName = execSoundexLuceneQueryByNames(false, name1, birthyear, includeVoided, gender);
-		secondNamesMatch.retainAll(secondNamesMatchFirstName);
-		people.addAll(secondNamesMatch);
-		secondNamesMatch.clear();
-		
-		// Executed Filter Logic:
-		// People who have given_name matching n2, second name matching n1, and another part of second name matching “”
-		firstNamesMatchSecondName = execSoundexLuceneQueryByNames(true, name2, birthyear, includeVoided, gender);
-		secondNamesMatch = execSoundexLuceneQueryByNames(false, "", birthyear, includeVoided, gender);
-		
-		firstNamesMatchSecondName.retainAll(secondNamesMatchFirstName);
-		firstNamesMatchSecondName.retainAll(secondNamesMatch);
-		people.addAll(firstNamesMatchSecondName);
-		
 		return people;
 	}
 	
 	
 	/**
-	 * This method executes a Lucene search on persons based on the soundex filter with three name elements given
+	 * This method executes a Lucene search on persons based on the soundex filter with two search names given
 	 *
-	 * @param name1 basically represents the first name to be searched for in a person name
-	 * @param name2 basically represents the middle name to be searched for in a person name	 
-	 * @param name3 basically represents the family name to be searched for in a person name
+	 * @param name1 firstName to be searched
+	 * @param name2 last name to be searched 
 	 * @param birthyear the birthyear the searched person should have 
 	 * @param includeVoided true if voided person should be included 
 	 * @param gender of the person to search for 
-	 * @return the set of Persons that match the search criteria 
+	 * @return the set of Persons that match the search criteria
 	 */
-	private Set<Person> executeSoundexThreePersonNamesQuery(String name1, String name2, String name3, Integer birthyear, boolean includeVoided , String gender) {
```

Comment:
Could you also remind me of why we are replacing the three names function with a two names one?